tests/expressions_case/tests.py

```diff
@@ -1243,3 +1243,31 @@ def test_conditional_aggregation_example(self):
             ),
             {'regular': 2, 'gold': 1, 'platinum': 3}
         )
+
+    def test_annotation_with_values_and_when(self):
```

Comment:
You can grep the docs to find the examples. At least some are in `docs/ref/models/conditional-expressions.txt`. I don't mind adding more tests, but it's a bit confusing to add them as part of a commit which fixes a different bug, so for clarity please make it a separate commit at least. Thanks!